### PR TITLE
Creating documentation

### LINKED ISSUE #60: Create Documentation
Build the documentation on GitHub Pages (should also run and render examples if possible).

### PR DESCRIPTION
**Pull request by elbeejay**
PR to create the documentation for this project and migrate it from readthedocs to GitHub Pages (to close #60). 

This largely follows the format we developed for the GOSTurban project, this consistency in documentation strategy should help reduce the maintenance burden in the future. 

This PR itself doesn't do a ton for formatting or editing of content, focus was on getting the existing content to execute and render in documentation and connecting up the CI etc. I think future PRs will be needed to clean up and improve the content itself (e.g., #62). I'd suggest we merge this so that the team can easily view the hosted documentation and open up issues, make suggestions, propose revisions, etc.

### CODE REVIEW

**Comment by elbeejay** [comment]
Should also close #22 and will close #1.

File: docs/api/GOSTnets.rst
Comment (g4brielvs) [inline docs/api/GOSTnets.rst:1]: @elbeejay This file should be generated by `sphinx-build`, so we ignore it from version control.

File: docs/api/modules.rst
Comment (g4brielvs) [inline docs/api/modules.rst:1]: @elbeejay This file should be generated by `sphinx-build`, so we ignore it from version control.

File: Tutorials/Step 1 - Extract From OSM.ipynb
Comment (g4brielvs) [inline Tutorials/Step 1 - Extract From OSM.ipynb:1]: @elbeejay Remove the note on the top as a Markdown header. We can use an admonition instead.

**Review by g4brielvs — CHANGES_REQUESTED** [review]

**Review by g4brielvs — APPROVED** [review]

**Pull request merged**